homeassistant/components/awair/config_flow.py

```diff
@@ -79,7 +166,27 @@ async def async_step_reauth_confirm(
             errors=errors,
         )
 
-    async def _check_connection(self, access_token: str):
+    async def _check_local_connection(self, device_address: str):
+        """Check the access token is valid."""
+        session = async_get_clientsession(self.hass)
+        awair = AwairLocal(session=session, device_addrs=[device_address])
+
+        try:
+            devices = await awair.devices()
+            if not devices:
+                return (None, "no_devices")
```

Comment:
Why do we need to call `devices` if we are targeting a single device via a local connection ?

Reply:
This seems to be the canonical way to interact with `python_awair`. The library is built around working with multiple devices. That being said, I was able to change and simplify this to just use a single device